gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -1194,6 +1217,11 @@ public InstantiatingGrpcChannelProvider build() {
                 ComputeEngineCredentials.GoogleAuthTransport.MTLS,
                 ComputeEngineCredentials.BindingEnforcement.ON);
       }
+      if (isDirectPathBoundTokenEnabled()) {
+        this.altsCallCredentials =
+            createHardBoundTokensCallCredentials(
+                ComputeEngineCredentials.GoogleAuthTransport.ALTS, null);
+      }
```

Comment:
In the future if allowedHardBoundTokenTypes contains both MTLS and ALTS with S2A enabled, it's possible that both mtls and alts credentials would be non-null? I'm assuming this is becuase of directpath only being determined when we create the channel.

Reply:
Yes we don't want these two credentials to interfere with each other, even though it's mostly one or the other when the client makes the connection. So they can co-exist.

Reply:
cc @rmehta19

Reply:
> even though it's mostly one or the other

Can you elaborate what you mean by mostly? If there is a scenario that both need to exist, then we'll need to have both and I think this looks fine

Reply:
I tried to explain in the other comment. Let's resolve this there.